adam_core/coordinates/tests/test_transforms_cometary.py

```diff
@@ -0,0 +1,174 @@
+import numpy as np
+from astropy import units as u
+
+from ..transform import cartesian_to_cometary, cometary_to_cartesian
+
+
+def test_cometary_to_cartesian_elliptical(orbital_elements):
+    # Test cometary_to_cartesian (vmapped) against the expected cartesian elements
+    # of a series of sample orbital elements.
+
+    # Limit to elliptical orbits
+    orbital_elements = orbital_elements[orbital_elements["e"] < 1]
+
+    cometary_elements = orbital_elements[
+        ["q", "e", "incl", "Omega", "w", "tp_mjd"]
+    ].values
+    cartesian_elements_expected = orbital_elements[
+        ["x", "y", "z", "vx", "vy", "vz"]
+    ].values
+    t0_mjd = orbital_elements["mjd_tdb"].values
+
+    cartesian_elements_actual = cometary_to_cartesian(cometary_elements, t0_mjd)
+    diff = cartesian_elements_actual - cartesian_elements_expected
+
+    # Calculate offset in position in mm
+    r_diff = np.linalg.norm(diff[:, :3], axis=1) * u.au.to(u.mm)
```

Comment:
I don't have a solution, because ndarray is likely the primitive we want to do most of this in, but I find the select by column index hard to follow and I worry about how brittle it is.